sed/core/processor.py

```diff
@@ -1545,9 +1765,23 @@ def save_delay_calibration(
         if filename is None:
             filename = "sed_config.yaml"
 
+        if len(self.dc.calibration) == 0:
+            raise ValueError("No delay calibration parameters to save!")
+        calibration = {}
+        for key, value in self.dc.calibration.items():
+            if key == "datafile":
+                calibration[key] = value
+            elif key in ["adc_range", "delay_range", "delay_range_mm"]:
+                calibration[key] = [float(i) for i in value]
+            else:
+                calibration[key] = float(value)
+
+        if "creation_date" not in calibration:
```

Comment:
I just looked at this one example of save_delay_calibration and can't seem to get the intended results. 

Firstly, the self.dc.calibration shows empty after running calibrate_delay_axis (using tutorial notebook 2) but no ValueError is raised.
And hence also the creation_date is not being stored in the config file. Actually tried for others and seems to be the same problem. Is it something I am doing wrong?

Reply:
What exactly did you do/check? The calibration is stored as processor.dc.calibration.
See lines 119 ff. and 193 in delay.py
They are only written to the config if you call the respective save_... function in the processor

Reply:
This is what I get:
![image](https://github.com/OpenCOMPES/sed/assets/53396064/e8da6bb2-9ca0-4adb-98d7-2eb95f2d7ef5)

Reply:
![image](https://github.com/OpenCOMPES/sed/assets/10053031/5d4cff9c-7d1a-4d83-b0f7-94f70bd91cde)

I am getting this

Reply:
I can't reproduce this. Are you sure you are on the right branch? Maybe try to debug and see if you come by the lines of code I indicated.

Reply:
@zain-sohail Do you need further input to test/debug? I tested once more and still cannot reproduce what you report.

Reply:
Sorry for late reply. There was some problem with the install and I managed to check it and everything looks good.